src/test/components/Viewport.test.js

```diff
@@ -0,0 +1,601 @@
+    const { view } = setup();
+    expect(view).toMatchSnapshot();
+    // Trigger any unmounting behavior handlers, just make sure it doesn't
+    // throw any errors.
+    view.unmount();
+  });
+
+  it('renders the wrapped chart component', () => {
+    const { view } = setup();
+    expect(view.find('#dummy-chart').length).toBe(1);
+  });
+
+  it('provides a viewport', () => {
+    const { getChartProps } = setup();
+    const { viewport } = getChartProps();
+    expect(viewport).toEqual({
+      containerWidth: COMPONENT_WIDTH,
+      containerHeight: COMPONENT_HEIGHT,
+      viewportLeft: 0,
+      viewportRight: 1,
+      viewportTop: 0,
+      viewportBottom: COMPONENT_HEIGHT,
+      isDragging: false,
+      isSizeSet: true,
+      // Just pass the function through.
+      moveViewport: viewport.moveViewport,
+    });
+  });
+
+  describe('scrolling hint', function() {
+    it('can show a shift scrolling hint', function() {
+      jest.useFakeTimers();
+      const { view, scroll } = setup();
+      const isTimerVisible = () =>
+        !view.find('.chartViewportShiftScroll').hasClass('hidden');
+
+      // No hint is shown.
+      expect(isTimerVisible()).toBe(false);
+
+      // Scroll a bit to show the menu
+      scroll({ deltaY: 10 });
+      expect(isTimerVisible()).toBe(true);
+
+      // Run the setTimeout, the menu should disappear.
+      jest.runAllTimers();
+      view.update();
+      expect(isTimerVisible()).toBe(false);
+    });
+
+    it('will not show a shift scrolling hint after zooming once', function() {
+      const { view, scroll } = setup();
+      const isTimerVisible = () =>
+        !view.find('.chartViewportShiftScroll').hasClass('hidden');
+
+      // No hint is shown at the beginning.
+      expect(isTimerVisible()).toBe(false);
+
+      // Zoom in a bit.
+      scroll({ deltaY: 10, shiftKey: true });
+      expect(isTimerVisible()).toBe(false);
+
+      // Now scroll, no hint should show.
+      scroll({ deltaY: 10 });
+      expect(isTimerVisible()).toBe(false);
+    });
+  });
+
+  describe('scrolling up and down', function() {
+    it('scrolls the viewport down using the mousewheel', () => {
+      const { view, getChartProps } = setup();
+      expect(getChartProps().viewport).toMatchObject({
+        viewportLeft: 0,
+        viewportRight: 1,
+        viewportTop: 0,
+        viewportBottom: COMPONENT_HEIGHT,
+      });
+      const deltaY = 10;
+      view.find('.chartViewport').simulate('wheel', getMouseEvent({ deltaY }));
+      expect(getChartProps().viewport).toMatchObject({
+        viewportLeft: 0,
+        viewportRight: 1,
+        viewportTop: deltaY,
+        viewportBottom: COMPONENT_HEIGHT + deltaY,
+      });
+    });
+
+    it('cannot scroll past the bottom of the viewport', () => {
+      const { view, getChartProps } = setup();
+      const deltaY = 1000;
+      view.find('.chartViewport').simulate('wheel', getMouseEvent({ deltaY }));
+      view.update();
```

Comment:
nit: you can use your `scroll` utility function here
(and same comment for the other tests below)

Reply:
here you can even use `scrollAndGetViewport` actually (unless we remove it, see the comment below)

Reply:
Thanks, I wrote these before I'd created that util. That's much cleaner.